### PR TITLE
adds safeAreaInsets to useViewer hook for the homepage

### PR DESCRIPTION
**Pull request by dylsteck**
[Per the linked ticket](https://github.com/warpcast/scores-frame/issues/38), this PR upgrades `@farcaster/frame-sdk` to v0.0.21 and adds safeAreaInsets to useViewer so that the home page can use it

### CODE REVIEW

**Review by goksu — DISMISSED** [review]
Thank you!

**Comment by goksu** [comment]
Closes warpcast/scores-frame#38

File: src/app/page.tsx [hunk 1]
```diff
@@ -69,7 +69,12 @@ export default function Home() {
   }, [lastWeeksSummary.rank]);
 
   return (
-    <div className="w-full h-full space-y-4 pb-4 px-4">
+    <div className="w-full h-full space-y-4 pb-4 px-4" style={ safeAreaInsets ? {
+      marginTop: safeAreaInsets.top,
+      marginBottom: safeAreaInsets.bottom,
```
Comment (pugson) [inline src/app/page.tsx:74]: margin causes layout shift. i would suggest using padding for all of these

Comment (dylsteck) [reply src/app/page.tsx]: thanks for the heads up @pugson!

**Pull request merged**